src/main/java/uk/gov/hmcts/reform/ccd/service/CaseFinderService.java

```diff
@@ -133,8 +133,8 @@ private boolean isCaseFamilyRetainable(final CaseFamily caseFamily) {
         final Set<Long> caseFamiliesToRetainCaseIds = getCaseFamiliesToRetainCaseIds(caseFamiliesToRetain);
 
         return caseFamilies.stream()
-            .filter(distinctByKey(caseFamily -> caseFamily.getRootAncestor().getId()))
-            .filter(caseFamily -> !caseFamiliesToRetainCaseIds.contains(caseFamily.getRootAncestor().getId()))
+            .filter(distinctByKey(caseFamily -> caseFamily.getRootCase().getId()))
+            .filter(caseFamily -> !caseFamiliesToRetainCaseIds.contains(caseFamily.getRootCase().getId()))
```

Comment:
Sonar reports that this line is only partially covered by unit tests.

Reply:
Fixed by commit 910f99ae55f163bb30f100e0e585aeafedd99c35